Fails to build

HIE rarely builds, this time I get:

Invocation was: `cabal new-build --jobs=2`

```
Failed to build ghc-exactprint-0.5.8.1.
Build log (
/home/maerwald/.cabal/logs/ghc-8.4.4/ghc-exactprint-0.5.8.1-b0c9671cc599593cc161b42a2e796e2f87fc0a6a4387d1e169be81b2cc9fc8ee.log
):
Configuring library for ghc-exactprint-0.5.8.1..
Preprocessing library for ghc-exactprint-0.5.8.1..
Building library for ghc-exactprint-0.5.8.1..
[ 1 of 13] Compiling Language.Haskell.GHC.ExactPrint.Types ( src/Language/Haskell/GHC/ExactPrint/Types.hs, dist/build/Language/Haskell/GHC/ExactPrint/Types.o )
[ 2 of 13] Compiling Language.Haskell.GHC.ExactPrint.Lookup ( src/Language/Haskell/GHC/ExactPrint/Lookup.hs, dist/build/Language/Haskell/GHC/ExactPrint/Lookup.o )
[ 3 of 13] Compiling Language.Haskell.GHC.ExactPrint.AnnotateTypes ( src/Language/Haskell/GHC/ExactPrint/AnnotateTypes.hs, dist/build/Language/Haskell/GHC/ExactPrint/AnnotateTypes.o )
[ 4 of 13] Compiling Language.Haskell.GHC.ExactPrint.Utils ( src/Language/Haskell/GHC/ExactPrint/Utils.hs, dist/build/Language/Haskell/GHC/ExactPrint/Utils.o )
[ 5 of 13] Compiling Language.Haskell.GHC.ExactPrint.Transform ( src/Language/Haskell/GHC/ExactPrint/Transform.hs, dist/build/Language/Haskell/GHC/ExactPrint/Transform.o )
[ 6 of 13] Compiling Language.Haskell.GHC.ExactPrint.Preprocess ( src/Language/Haskell/GHC/ExactPrint/Preprocess.hs, dist/build/Language/Haskell/GHC/ExactPrint/Preprocess.o )
[ 7 of 13] Compiling Language.Haskell.GHC.ExactPrint.Annotater ( src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs, dist/build/Language/Haskell/GHC/ExactPrint/Annotater.o )

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:290:9: error:
    Not in scope: data constructor ‘GHC.XIE’
    Perhaps you meant one of these:
      ‘GHC.ABE’ (imported from GHC), ‘GHC.IL’ (imported from BasicTypes),
      ‘GHC.GRE’ (imported from RdrName)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XIE’.
    |
290 |         GHC.XIE x -> error $ "got XIE for :" ++ showGhc x
    |         ^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:446:13: error:
    Not in scope: data constructor ‘GHC.XImportDecl’
    Perhaps you meant ‘GHC.ImportDecl’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XImportDecl’.
    |
446 |  markAST _ (GHC.XImportDecl x) = error $ "got XImportDecl for:" ++ showGhc x
    |             ^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:472:7: error:
    Not in scope: data constructor ‘GHC.XHsDecl’
    Perhaps you meant one of these:
      variable ‘GHC.isDecl’ (imported from GHC),
      ‘GHC.FamDecl’ (imported from GHC), ‘GHC.HsVect’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XHsDecl’.
    |
472 |       GHC.XHsDecl x  -> error $ "got XHsDecl for:" ++ showGhc x
    |       ^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:485:14: error:
    Not in scope: data constructor ‘GHC.XRoleAnnotDecl’
    Perhaps you meant one of these:
      ‘GHC.RoleAnnotDecl’ (imported from GHC),
      ‘GHC.RoleAnnotD’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XRoleAnnotDecl’.
    |
485 |   markAST _ (GHC.XRoleAnnotDecl x) = error $ "got XRoleAnnotDecl for:" ++ showGhc x
    |              ^^^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:501:14: error:
    Not in scope: data constructor ‘GHC.XSpliceDecl’
    Perhaps you meant one of these:
      ‘GHC.SpliceDecl’ (imported from GHC),
      variable ‘GHC.pprSpliceDecl’ (imported from GHC),
      ‘GHC.SpliceD’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XSpliceDecl’.
    |
501 |   markAST _ (GHC.XSpliceDecl x) = error $ "got XSpliceDecl for:" ++ showGhc x
    |              ^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:511:14: error:
    Not in scope: data constructor ‘GHC.XRuleDecls’
    Perhaps you meant variable ‘GHC.runDecls’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XRuleDecls’.
    |
511 |   markAST _ (GHC.XRuleDecls x) = error $ "got XRuleDecls for:" ++ showGhc x
    |              ^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:531:14: error:
    Not in scope: data constructor ‘GHC.XRuleDecl’
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XRuleDecl’.
    |
531 |   markAST _ (GHC.XRuleDecl x) = error $ "got XRuleDecl for:" ++ showGhc x
    |              ^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:564:14: error:
    Not in scope: data constructor ‘GHC.XRuleBndr’
    Perhaps you meant ‘GHC.RuleBndr’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XRuleBndr’.
    |
564 |   markAST _ (GHC.XRuleBndr x) = error $ "got XRuleBndr for:" ++ showGhc x
    |              ^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:571:31: error:
    Not in scope: data constructor ‘GHC.XHsImplicitBndrs’
    Perhaps you meant one of these:
      variable ‘GHC.mkHsImplicitBndrs’ (imported from GHC),
      variable ‘GHC.hsImplicitBody’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XHsImplicitBndrs’.
    |
571 | markLHsSigWcType (GHC.HsWC _ (GHC.XHsImplicitBndrs _)) = error "markLHsSigWcType extension hit"
    |                               ^^^^^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:572:19: error:
    Not in scope: data constructor ‘GHC.XHsWildCardBndrs’
    Perhaps you meant variable ‘GHC.mkHsWildCardBndrs’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XHsWildCardBndrs’.
    |
572 | markLHsSigWcType (GHC.XHsWildCardBndrs _)              = error "markLHsSigWcType extension hit"
    |                   ^^^^^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:590:15: error:
    Not in scope: data constructor ‘GHC.XAnnDecl’
    Perhaps you meant one of these:
      ‘GHC.AnnRec’ (imported from GHC), ‘GHC.SynDecl’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XAnnDecl’.
    |
590 |    markAST _ (GHC.XAnnDecl x) = error $ "got XAnnDecl for:" ++ showGhc x
    |               ^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:600:15: error:
    Not in scope: data constructor ‘GHC.XWarnDecls’
    Perhaps you meant variable ‘GHC.runDecls’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XWarnDecls’.
    |
600 |    markAST _ (GHC.XWarnDecls x) = error $ "got XWarnDecls for:" ++ showGhc x
    |               ^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:613:15: error:
    Not in scope: data constructor ‘GHC.XWarnDecl’
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XWarnDecl’.
    |
613 |    markAST _ (GHC.XWarnDecl x) = error $ "got XWarnDecl for:" ++ showGhc x
    |               ^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:646:37: error:
    Not in scope: data constructor ‘GHC.XHsImplicitBndrs’
    Perhaps you meant one of these:
      variable ‘GHC.mkHsImplicitBndrs’ (imported from GHC),
      variable ‘GHC.hsImplicitBody’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XHsImplicitBndrs’.
    |
646 |   markAST _ (GHC.ForeignImport _ _ (GHC.XHsImplicitBndrs _) _) = error "markAST ForeignDecl hit extenstion"
    |                                     ^^^^^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:647:37: error:
    Not in scope: data constructor ‘GHC.XHsImplicitBndrs’
    Perhaps you meant one of these:
      variable ‘GHC.mkHsImplicitBndrs’ (imported from GHC),
      variable ‘GHC.hsImplicitBody’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XHsImplicitBndrs’.
    |
647 |   markAST _ (GHC.ForeignExport _ _ (GHC.XHsImplicitBndrs _) _) = error "markAST ForeignDecl hit extenstion"
    |                                     ^^^^^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:648:14: error:
    Not in scope: data constructor ‘GHC.XForeignDecl’
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XForeignDecl’.
    |
648 |   markAST _ (GHC.XForeignDecl _)                               = error "markAST ForeignDecl hit extenstion"
    |              ^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:720:43: error:
    Not in scope: data constructor ‘GHC.XHsImplicitBndrs’
    Perhaps you meant one of these:
      variable ‘GHC.mkHsImplicitBndrs’ (imported from GHC),
      variable ‘GHC.hsImplicitBody’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XHsImplicitBndrs’.
    |
720 |   markAST _ (GHC.DerivDecl _ (GHC.HsWC _ (GHC.XHsImplicitBndrs _)) _ _) = error "markAST DerivDecl hit extension"
    |                                           ^^^^^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:721:31: error:
    Not in scope: data constructor ‘GHC.XHsWildCardBndrs’
    Perhaps you meant variable ‘GHC.mkHsWildCardBndrs’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XHsWildCardBndrs’.
    |
721 |   markAST _ (GHC.DerivDecl _ (GHC.XHsWildCardBndrs _) _ _)              = error "markAST DerivDecl hit extension"
    |                               ^^^^^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:722:14: error:
    Not in scope: data constructor ‘GHC.XDerivDecl’
    Perhaps you meant one of these:
      ‘GHC.DerivDecl’ (imported from GHC),
      ‘GHC.DerivD’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XDerivDecl’.
    |
722 |   markAST _ (GHC.XDerivDecl _)                                          = error "markAST DerivDecl hit extension"
    |              ^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:731:14: error:
    Not in scope: data constructor ‘GHC.ViaStrategy’
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘ViaStrategy’.
    |
731 |   markAST _ (GHC.ViaStrategy (GHC.HsIB _ ty)) = do
    |              ^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:732:10: error:
    Not in scope: data constructor ‘GHC.AnnVia’
    Perhaps you meant one of these:
      ‘GHC.AnnVal’ (imported from GHC),
      ‘GHC.AnnVbar’ (imported from GHC), ‘GHC.AnnD’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘AnnVia’.
    |
732 |     mark GHC.AnnVia
    |          ^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:734:14: error:
    Not in scope: data constructor ‘GHC.ViaStrategy’
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘ViaStrategy’.
    |
734 |   markAST _ (GHC.ViaStrategy (GHC.XHsImplicitBndrs _))
    |              ^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:734:31: error:
    Not in scope: data constructor ‘GHC.XHsImplicitBndrs’
    Perhaps you meant one of these:
      variable ‘GHC.mkHsImplicitBndrs’ (imported from GHC),
      variable ‘GHC.hsImplicitBody’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XHsImplicitBndrs’.
    |
734 |   markAST _ (GHC.ViaStrategy (GHC.XHsImplicitBndrs _))
    |                               ^^^^^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:748:14: error:
    Not in scope: data constructor ‘GHC.XDefaultDecl’
    Perhaps you meant ‘GHC.DefaultDecl’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XDefaultDecl’.
    |
748 |   markAST _ (GHC.XDefaultDecl x) = error $ "got XDefaultDecl for:" ++ showGhc x
    |              ^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:757:14: error:
    Not in scope: data constructor ‘GHC.XInstDecl’
    Perhaps you meant ‘GHC.ClsInstDecl’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XInstDecl’.
    |
757 |   markAST _ (GHC.XInstDecl x) = error $ "got XInstDecl for:" ++ showGhc x
    |              ^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:805:33: error:
    Not in scope: data constructor ‘GHC.XHsImplicitBndrs’
    Perhaps you meant one of these:
      variable ‘GHC.mkHsImplicitBndrs’ (imported from GHC),
      variable ‘GHC.hsImplicitBody’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XHsImplicitBndrs’.
    |
805 |   markAST _ (GHC.ClsInstDecl _ (GHC.XHsImplicitBndrs _) _ _ _ _ _) = error "extension hit for ClsInstDecl"
    |                                 ^^^^^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:806:14: error:
    Not in scope: data constructor ‘GHC.XClsInstDecl’
    Perhaps you meant one of these:
      ‘GHC.ClsInstDecl’ (imported from GHC),
      ‘GHC.ClsInstD’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XClsInstDecl’.
    |
806 |   markAST _ (GHC.XClsInstDecl _)                                   = error "extension hit for ClsInstDecl"
    |              ^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:818:33: error:
    Not in scope: data constructor ‘GHC.XHsImplicitBndrs’
    Perhaps you meant one of these:
      variable ‘GHC.mkHsImplicitBndrs’ (imported from GHC),
      variable ‘GHC.hsImplicitBody’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XHsImplicitBndrs’.
    |
818 |   markAST _ (GHC.TyFamInstDecl (GHC.XHsImplicitBndrs _)) = error "extension hit for TyFamInstDecl"
    |                                 ^^^^^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:830:13: error:
    Not in scope: data constructor ‘GHC.XFamEqn’
    Perhaps you meant ‘GHC.FamEqn’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XFamEqn’.
    |
830 | markFamEqn (GHC.XFamEqn _) = error "got XFamEqn"
    |             ^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:855:36: error:
    Not in scope: ‘GHC.noExt’
    Perhaps you meant ‘GHC.noExpr’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘noExt’.
    |
855 |     markDataDefn l (GHC.HsDataDefn GHC.noExt nd (GHC.noLoc []) typ _mk cons mderivs)
    |                                    ^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:877:47: error:
    Not in scope: data constructor ‘GHC.XHsDataDefn’
    Perhaps you meant one of these:
      ‘GHC.HsDataDefn’ (imported from GHC),
      variable ‘GHC.tcdDataDefn’ (imported from GHC),
      variable ‘GHC.isDataDecl’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XHsDataDefn’.
    |
877 |              (GHC.HsIB _ (GHC.FamEqn _ _ _ _ (GHC.XHsDataDefn _))))
    |                                               ^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:879:47: error:
    Not in scope: data constructor ‘GHC.XFamEqn’
    Perhaps you meant ‘GHC.FamEqn’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XFamEqn’.
    |
879 |   markAST _ (GHC.DataFamInstDecl (GHC.HsIB _ (GHC.XFamEqn _)))
    |                                               ^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:881:35: error:
    Not in scope: data constructor ‘GHC.XHsImplicitBndrs’
    Perhaps you meant one of these:
      variable ‘GHC.mkHsImplicitBndrs’ (imported from GHC),
      variable ‘GHC.hsImplicitBody’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XHsImplicitBndrs’.
    |
881 |   markAST _ (GHC.DataFamInstDecl (GHC.XHsImplicitBndrs _))
    |                                   ^^^^^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:968:31: error:
    Not in scope: data constructor ‘GHC.XMatchGroup’
    Perhaps you meant variable ‘GHC.mkMatchGroup’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XMatchGroup’.
    |
968 |   markAST _ (GHC.FunBind _ _ (GHC.XMatchGroup _) _ _)
    |                               ^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:970:31: error:
    Not in scope: data constructor ‘GHC.XGRHSs’
    Perhaps you meant one of these:
      ‘GHC.GRHSs’ (imported from GHC), ‘GHC.GRHS’ (imported from GHC),
      variable ‘GHC.pprGRHSs’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XGRHSs’.
    |
970 |   markAST _ (GHC.PatBind _ _ (GHC.XGRHSs _) _)
    |                               ^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:972:32: error:
    Not in scope: data constructor ‘GHC.XPatSynBind’
    Perhaps you meant one of these:
      ‘GHC.PatSynBind’ (imported from GHC),
      variable ‘GHC.mkPatSynBind’ (imported from GHC),
      ‘GHC.PatBind’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XPatSynBind’.
    |
972 |   markAST _ (GHC.PatSynBind _ (GHC.XPatSynBind _))
    |                                ^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:974:14: error:
    Not in scope: data constructor ‘GHC.XHsBindsLR’
    Perhaps you meant ‘GHC.AbsBinds’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XHsBindsLR’.
    |
974 |   markAST _ (GHC.XHsBindsLR _)
    |              ^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:989:14: error:
    Not in scope: data constructor ‘GHC.XIPBind’
    Perhaps you meant one of these:
      ‘GHC.IPBind’ (imported from GHC),
      ‘GHC.IPBinds’ (imported from GHC),
      ‘GHC.FunBind’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XIPBind’.
    |
989 |   markAST _ (GHC.XIPBind x) = error $ "got XIPBind for:" ++ showGhc x
    |              ^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1052:31: error:
    Not in scope: data constructor ‘GHC.XGRHSs’
    Perhaps you meant one of these:
      ‘GHC.GRHSs’ (imported from GHC), ‘GHC.GRHS’ (imported from GHC),
      variable ‘GHC.pprGRHSs’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XGRHSs’.
     |
1052 |   markAST _ (GHC.Match _ _ _ (GHC.XGRHSs _))
     |                               ^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1054:14: error:
    Not in scope: data constructor ‘GHC.XMatch’
    Perhaps you meant one of these:
      ‘GHC.Match’ (imported from GHC),
      variable ‘GHC.gcatch’ (imported from GHC),
      variable ‘GHC.mkMatch’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XMatch’.
     |
1054 |   markAST _ (GHC.XMatch _)
     |              ^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1077:14: error:
    Not in scope: data constructor ‘GHC.XGRHS’
    Perhaps you meant one of these:
      ‘GHC.GRHS’ (imported from GHC), ‘GHC.GRHSs’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XGRHS’.
     |
1077 |   markAST _ (GHC.XGRHS x) = error $ "got XGRHS for:" ++ showGhc x
     |              ^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1167:33: error:
    Not in scope: data constructor ‘GHC.XHsImplicitBndrs’
    Perhaps you meant one of these:
      variable ‘GHC.mkHsImplicitBndrs’ (imported from GHC),
      variable ‘GHC.hsImplicitBody’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XHsImplicitBndrs’.
     |
1167 |   markAST _ (GHC.PatSynSig _ _ (GHC.XHsImplicitBndrs _))
     |                                 ^^^^^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1169:36: error:
    Not in scope: data constructor ‘GHC.XHsImplicitBndrs’
    Perhaps you meant one of these:
      variable ‘GHC.mkHsImplicitBndrs’ (imported from GHC),
      variable ‘GHC.hsImplicitBody’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XHsImplicitBndrs’.
     |
1169 |   markAST _ (GHC.ClassOpSig _ _ _ (GHC.XHsImplicitBndrs _))
     |                                    ^^^^^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1171:28: error:
    Not in scope: data constructor ‘GHC.XFixitySig’
    Perhaps you meant one of these:
      ‘GHC.FixitySig’ (imported from GHC),
      variable ‘GHC.isFixityLSig’ (imported from GHC),
      ‘GHC.Fixity’ (imported from BasicTypes)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XFixitySig’.
     |
1171 |   markAST _ (GHC.FixSig _ (GHC.XFixitySig _))
     |                            ^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1173:14: error:
    Not in scope: data constructor ‘GHC.XSig’
    Perhaps you meant one of these:
      ‘GHC.SigD’ (imported from GHC), ‘GHC.IdSig’ (imported from GHC),
      ‘GHC.NoSig’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XSig’.
     |
1173 |   markAST _ (GHC.XSig _)
     |              ^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1180:17: error:
    Not in scope: data constructor ‘GHC.XHsImplicitBndrs’
    Perhaps you meant one of these:
      variable ‘GHC.mkHsImplicitBndrs’ (imported from GHC),
      variable ‘GHC.hsImplicitBody’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XHsImplicitBndrs’.
     |
1180 | markLHsSigType (GHC.XHsImplicitBndrs x) = error $ "got XHsImplicitBndrs for:" ++ showGhc x
     |                 ^^^^^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1190:32: error:
    Not in scope: ‘GHC.hsTypeNeedsParens’
    Perhaps you meant one of these:
      ‘GHC.hsExprNeedsParens’ (imported from GHC),
      ‘GHC.hsPatNeedsParens’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘hsTypeNeedsParens’.
     |
1190 |           [GHC.HsIB _ t] -> if GHC.hsTypeNeedsParens GHC.appPrec (GHC.unLoc t)
     |                                ^^^^^^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1190:54: error:
    Not in scope: ‘GHC.appPrec’
    Perhaps you meant one of these:
      data constructor ‘GHC.TopPrec’ (imported from BasicTypes),
      ‘GHC.pprPrec’ (imported from Outputable),
      data constructor ‘GHC.FunPrec’ (imported from BasicTypes)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘appPrec’.
     |
1190 |           [GHC.HsIB _ t] -> if GHC.hsTypeNeedsParens GHC.appPrec (GHC.unLoc t)
     |                                                      ^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1230:15: error:
    Not in scope: data constructor ‘GHC.XTyVarBndr’
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XTyVarBndr’.
     |
1230 |   markAST _l (GHC.XTyVarBndr x) = error $ "got XTyVarBndr for:" ++ showGhc x
     |               ^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1306:17: error:
    Not in scope: data constructor ‘GHC.HsStarTy’
    Perhaps you meant one of these:
      ‘GHC.HsStrTy’ (imported from GHC),
      ‘GHC.HsParTy’ (imported from GHC),
      ‘GHC.HsStatic’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘HsStarTy’.
     |
1306 |     markType l (GHC.HsStarTy _ isUnicode) = do
     |                 ^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1365:17: error:
    Not in scope: data constructor ‘GHC.XHsType’
    Perhaps you meant one of these:
      ‘GHC.AnnType’ (imported from GHC),
      ‘GHC.NewType’ (imported from GHC), ‘GHC.IEType’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XHsType’.
     |
1365 |     markType _ (GHC.XHsType x) = error $ "got XHsType for:" ++ showGhc x
     |                 ^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1421:8: error:
    Not in scope: data constructor ‘GHC.XSplice’
    Perhaps you meant one of these:
      ‘GHC.SpliceD’ (imported from GHC),
      ‘GHC.HsSpliceE’ (imported from GHC),
      ‘GHC.HsSpliced’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XSplice’.
     |
1421 |       (GHC.XSplice x) -> error $ "got XSplice for:" ++ showGhc x
     |        ^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1434:14: error:
    Not in scope: data constructor ‘GHC.XConDeclField’
    Perhaps you meant one of these:
      ‘GHC.ConDeclField’ (imported from GHC),
      variable ‘GHC.pprConDeclFields’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XConDeclField’.
     |
1434 |   markAST _ (GHC.XConDeclField x) = error $ "got XConDeclField for:" ++ showGhc x
     |              ^^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1443:14: error:
    Not in scope: data constructor ‘GHC.XFieldOcc’
    Perhaps you meant one of these:
      ‘GHC.FieldOcc’ (imported from GHC),
      variable ‘GHC.mkFieldOcc’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XFieldOcc’.
     |
1443 |   markAST _ (GHC.XFieldOcc x) = error $ "got XFieldOcc for:" ++ showGhc x
     |              ^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1449:32: error:
    Not in scope: ‘GHC.unpackHDS’
    Perhaps you meant ‘GHC.unpackFS’ (imported from FastString)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘unpackHDS’.
     |
1449 |     markExternal l GHC.AnnVal (GHC.unpackHDS s)
     |                                ^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1529:18: error:
    Not in scope: data constructor ‘GHC.SigPat’
    Perhaps you meant one of these:
      ‘GHC.Signal’ (imported from GHC), ‘GHC.LitPat’ (imported from GHC),
      ‘GHC.SigPatIn’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘SigPat’.
     |
1529 |       markPat _ (GHC.SigPat ty pat) = do
     |                  ^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1537:18: error:
    Not in scope: data constructor ‘GHC.XPat’
    Perhaps you meant one of these:
      ‘GHC.NPat’ (imported from GHC), ‘GHC.AsPat’ (imported from GHC),
      ‘GHC.CoPat’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XPat’.
     |
1537 |       markPat _ (GHC.XPat x) = error $ "got XPat for:" ++ showGhc x
     |                  ^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1559:6: error:
    Not in scope: data constructor ‘GHC.XLit’
    Perhaps you meant one of these:
      variable ‘GHC.sLit’ (imported from FastString),
      ‘GHC.HsLit’ (imported from GHC),
      variable ‘GHC.fsLit’ (imported from FastString)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XLit’.
     |
1559 |     (GHC.XLit x) -> error $ "got XLit for:" ++ showGhc x
     |      ^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1639:14: error:
    Not in scope: data constructor ‘GHC.XHsImplicitBndrs’
    Perhaps you meant one of these:
      variable ‘GHC.mkHsImplicitBndrs’ (imported from GHC),
      variable ‘GHC.hsImplicitBody’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XHsImplicitBndrs’.
     |
1639 |   markAST _ (GHC.XHsImplicitBndrs x) = error $ "got XHsImplicitBndrs for:" ++ showGhc x
     |              ^^^^^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1739:14: error:
    Not in scope: data constructor ‘GHC.XStmtLR’
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XStmtLR’.
     |
1739 |   markAST _ (GHC.XStmtLR x) = error $ "got XStmtLR for:" ++ showGhc x
     |              ^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1748:14: error:
    Not in scope: data constructor ‘GHC.XParStmtBlock’
    Perhaps you meant ‘GHC.ParStmtBlock’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XParStmtBlock’.
     |
1748 |   markAST _ (GHC.XParStmtBlock x) = error $ "got XParStmtBlock for:" ++ showGhc x
     |              ^^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1758:37: error:
    Not in scope: data constructor ‘GHC.ValBinds’
    Perhaps you meant one of these:
      ‘GHC.HsValBinds’ (imported from GHC),
      ‘GHC.ValBindsIn’ (imported from GHC),
      ‘GHC.VarBind’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘ValBinds’.
     |
1758 | markHsLocalBinds (GHC.HsValBinds _ (GHC.ValBinds _ binds sigs)) =
     |                                     ^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1766:37: error:
    Not in scope: data constructor ‘GHC.XValBindsLR’
    Perhaps you meant one of these:
      ‘GHC.ValBindsIn’ (imported from GHC),
      ‘GHC.HsValBinds’ (imported from GHC),
      ‘GHC.ValBindsOut’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XValBindsLR’.
     |
1766 | markHsLocalBinds (GHC.HsValBinds _ (GHC.XValBindsLR _)) = error "markHsLocalBinds:got extension"
     |                                     ^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1767:36: error:
    Not in scope: data constructor ‘GHC.XHsIPBinds’
    Perhaps you meant one of these:
      ‘GHC.HsIPBinds’ (imported from GHC),
      ‘GHC.IPBinds’ (imported from GHC),
      ‘GHC.AbsBinds’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XHsIPBinds’.
     |
1767 | markHsLocalBinds (GHC.HsIPBinds _ (GHC.XHsIPBinds _))   = error "markHsLocalBinds:got extension"
     |                                    ^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1768:19: error:
    Not in scope: data constructor ‘GHC.XHsLocalBindsLR’
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XHsLocalBindsLR’.
     |
1768 | markHsLocalBinds (GHC.XHsLocalBindsLR _)                = error "markHsLocalBinds:got extension"
     |                   ^^^^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:1777:19: error:
    Not in scope: data constructor ‘GHC.XMatchGroup’
    Perhaps you meant variable ‘GHC.mkMatchGroup’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XMatchGroup’.
     |
1777 | markMatchGroup _ (GHC.XMatchGroup x) = error $ "got XMatchGroup for:" ++ showGhc x
     |                   ^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2157:36: error:
    Not in scope: data constructor ‘GHC.XBracket’
    Perhaps you meant one of these:
      ‘GHC.HsBracket’ (imported from GHC),
      variable ‘GHC.gbracket’ (imported from GHC),
      variable ‘GHC.brackets’ (imported from Outputable)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XBracket’.
     |
2157 |       markExpr _ (GHC.HsBracket _ (GHC.XBracket _)) = error "markExpr got extension"
     |                                    ^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2158:19: error:
    Not in scope: data constructor ‘GHC.XExpr’
    Perhaps you meant one of these:
      ‘GHC.ExpBr’ (imported from GHC), ‘GHC.TExpBr’ (imported from GHC),
      ‘GHC.DoExpr’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XExpr’.
     |
2158 |       markExpr _ (GHC.XExpr _)                      = error "markExpr got extension"
     |                   ^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2165:16: error:
    Not in scope: data constructor ‘GHC.XHsWildCardBndrs’
    Perhaps you meant variable ‘GHC.mkHsWildCardBndrs’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XHsWildCardBndrs’.
     |
2165 | markLHsWcType (GHC.XHsWildCardBndrs x) = error $ "markLHsWcType got :" ++ showGhc x
     |                ^^^^^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2185:14: error:
    Not in scope: data constructor ‘GHC.XAmbiguousFieldOcc’
    Perhaps you meant one of these:
      variable ‘GHC.ambiguousFieldOcc’ (imported from GHC),
      variable ‘GHC.mkAmbiguousFieldOcc’ (imported from GHC),
      variable ‘GHC.unambiguousFieldOcc’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XAmbiguousFieldOcc’.
     |
2185 |   markAST _ (GHC.XAmbiguousFieldOcc x) = error $ "got XAmbiguousFieldOcc for:" ++ showGhc x
     |              ^^^^^^^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2204:14: error:
    Not in scope: data constructor ‘GHC.XTupArg’
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XTupArg’.
     |
2204 |   markAST _ (GHC.XTupArg x) = error $ "got XTupArg got:" ++ showGhc x
     |              ^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2210:14: error:
    Not in scope: data constructor ‘GHC.XCmdTop’
    Perhaps you meant ‘GHC.HsCmdTop’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XCmdTop’.
     |
2210 |   markAST _ (GHC.XCmdTop x) = error $ "got XCmdTop for:" ++ showGhc x
     |              ^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2296:14: error:
    Not in scope: data constructor ‘GHC.XCmd’
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XCmd’.
     |
2296 |   markAST _ (GHC.XCmd x) = error $ "got XCmd for:" ++ showGhc x
     |              ^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2391:31: error:
    Not in scope: data constructor ‘GHC.XLHsQTyVars’
    Perhaps you meant one of these:
      ‘GHC.HsTyVar’ (imported from GHC),
      variable ‘GHC.nlHsTyVar’ (imported from GHC),
      variable ‘GHC.classTyVars’ (imported from Class)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XLHsQTyVars’.
     |
2391 |   markAST _ (GHC.SynDecl _ _ (GHC.XLHsQTyVars _) _ _)
     |                               ^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2393:51: error:
    Not in scope: data constructor ‘GHC.XHsDataDefn’
    Perhaps you meant one of these:
      ‘GHC.HsDataDefn’ (imported from GHC),
      variable ‘GHC.tcdDataDefn’ (imported from GHC),
      variable ‘GHC.isDataDecl’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XHsDataDefn’.
     |
2393 |   markAST _ (GHC.DataDecl _ _ (GHC.HsQTvs _ _) _ (GHC.XHsDataDefn _))
     |                                                   ^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2395:32: error:
    Not in scope: data constructor ‘GHC.XLHsQTyVars’
    Perhaps you meant one of these:
      ‘GHC.HsTyVar’ (imported from GHC),
      variable ‘GHC.nlHsTyVar’ (imported from GHC),
      variable ‘GHC.classTyVars’ (imported from Class)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XLHsQTyVars’.
     |
2395 |   markAST _ (GHC.DataDecl _ _ (GHC.XLHsQTyVars _) _ _)
     |                                ^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2397:35: error:
    Not in scope: data constructor ‘GHC.XLHsQTyVars’
    Perhaps you meant one of these:
      ‘GHC.HsTyVar’ (imported from GHC),
      variable ‘GHC.nlHsTyVar’ (imported from GHC),
      variable ‘GHC.classTyVars’ (imported from Class)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XLHsQTyVars’.
     |
2397 |   markAST _ (GHC.ClassDecl _ _ _ (GHC.XLHsQTyVars _) _ _ _ _ _ _ _)
     |                                   ^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2399:14: error:
    Not in scope: data constructor ‘GHC.XTyClDecl’
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XTyClDecl’.
     |
2399 |   markAST _ (GHC.XTyClDecl _)
     |              ^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2447:22: error:
    Not in scope: data constructor ‘GHC.ViaStrategy’
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘ViaStrategy’.
     |
2447 |       Just (GHC.L _ (GHC.ViaStrategy{})) -> return ()
     |                      ^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2451:24: error:
    Not in scope: data constructor ‘GHC.ViaStrategy’
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘ViaStrategy’.
     |
2451 |       Just s@(GHC.L _ (GHC.ViaStrategy{})) -> markLocated s
     |                        ^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2454:14: error:
    Not in scope: data constructor ‘GHC.XHsDerivingClause’
    Perhaps you meant ‘GHC.HsDerivingClause’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XHsDerivingClause’.
     |
2454 |   markAST _ (GHC.XHsDerivingClause x) = error $ "got XHsDerivingClause for:" ++ showGhc x
     |              ^^^^^^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2494:8: error:
    Not in scope: data constructor ‘GHC.XFamilyResultSig’
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XFamilyResultSig’.
     |
2494 |       (GHC.XFamilyResultSig x) -> error $ "FamilyDecl:got XFamilyResultSig for:" ++ showGhc x
     |        ^^^^^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2514:36: error:
    Not in scope: data constructor ‘GHC.XLHsQTyVars’
    Perhaps you meant one of these:
      ‘GHC.HsTyVar’ (imported from GHC),
      variable ‘GHC.nlHsTyVar’ (imported from GHC),
      variable ‘GHC.classTyVars’ (imported from Class)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XLHsQTyVars’.
     |
2514 |   markAST _ (GHC.FamilyDecl _ _ _ (GHC.XLHsQTyVars _) _ _ _)
     |                                    ^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2516:14: error:
    Not in scope: data constructor ‘GHC.XFamilyDecl’
    Perhaps you meant one of these:
      ‘GHC.FamilyDecl’ (imported from GHC),
      variable ‘GHC.isFamilyDecl’ (imported from GHC),
      ‘GHC.FamDecl’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XFamilyDecl’.
     |
2516 |   markAST _ (GHC.XFamilyDecl _)
     |              ^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2525:14: error:
    Not in scope: data constructor ‘GHC.XFamilyResultSig’
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XFamilyResultSig’.
     |
2525 |   markAST _ (GHC.XFamilyResultSig x) = error $ "got XFamilyResultSig for:" ++ showGhc x
     |              ^^^^^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2542:14: error:
    Not in scope: data constructor ‘GHC.XHsImplicitBndrs’
    Perhaps you meant one of these:
      variable ‘GHC.mkHsImplicitBndrs’ (imported from GHC),
      variable ‘GHC.hsImplicitBody’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XHsImplicitBndrs’.
     |
2542 |   markAST _ (GHC.XHsImplicitBndrs x) = error $ "got XHsImplicitBndrs for:" ++ showGhc x
     |              ^^^^^^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2555:30: error:
    Not in scope: data constructor ‘GHC.XLHsQTyVars’
    Perhaps you meant one of these:
      ‘GHC.HsTyVar’ (imported from GHC),
      variable ‘GHC.nlHsTyVar’ (imported from GHC),
      variable ‘GHC.classTyVars’ (imported from Class)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XLHsQTyVars’.
     |
2555 |   markAST _ (GHC.FamEqn _ _ (GHC.XLHsQTyVars _) _ _)
     |                              ^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2557:14: error:
    Not in scope: data constructor ‘GHC.XFamEqn’
    Perhaps you meant ‘GHC.FamEqn’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XFamEqn’.
     |
2557 |   markAST _ (GHC.XFamEqn _)
     |              ^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2567:44: error:
    Not in scope: ‘GHC.unpackHDS’
    Perhaps you meant ‘GHC.unpackFS’ (imported from FastString)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘unpackHDS’.
     |
2567 |             (GHC.DocCommentNext ds)     -> GHC.unpackHDS ds
     |                                            ^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2568:44: error:
    Not in scope: ‘GHC.unpackHDS’
    Perhaps you meant ‘GHC.unpackFS’ (imported from FastString)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘unpackHDS’.
     |
2568 |             (GHC.DocCommentPrev ds)     -> GHC.unpackHDS ds
     |                                            ^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2569:44: error:
    Not in scope: ‘GHC.unpackHDS’
    Perhaps you meant ‘GHC.unpackFS’ (imported from FastString)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘unpackHDS’.
     |
2569 |             (GHC.DocCommentNamed _s ds) -> GHC.unpackHDS ds
     |                                            ^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2570:44: error:
    Not in scope: ‘GHC.unpackHDS’
    Perhaps you meant ‘GHC.unpackFS’ (imported from FastString)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘unpackHDS’.
     |
2570 |             (GHC.DocGroup _i ds)        -> GHC.unpackHDS ds
     |                                            ^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2592:17: error:
    Not in scope: data constructor ‘GHC.XHsDataDefn’
    Perhaps you meant one of these:
      ‘GHC.HsDataDefn’ (imported from GHC),
      variable ‘GHC.tcdDataDefn’ (imported from GHC),
      variable ‘GHC.isDataDecl’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XHsDataDefn’.
     |
2592 | markDataDefn _ (GHC.XHsDataDefn x) = error $ "got XHsDataDefn for:" ++ showGhc x
     |                 ^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2715:47: error:
    Not in scope: data constructor ‘GHC.XLHsQTyVars’
    Perhaps you meant one of these:
      ‘GHC.HsTyVar’ (imported from GHC),
      variable ‘GHC.nlHsTyVar’ (imported from GHC),
      variable ‘GHC.classTyVars’ (imported from Class)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XLHsQTyVars’.
     |
2715 |   markAST _ (GHC.ConDeclGADT _ _ (GHC.L _ _) (GHC.XLHsQTyVars _) _ _ _ _)
     |                                               ^^^^^^^^^^^^^^^

src-ghc86/Language/Haskell/GHC/ExactPrint/Annotater.hs:2717:14: error:
    Not in scope: data constructor ‘GHC.XConDecl’
    Perhaps you meant ‘GHC.SynDecl’ (imported from GHC)
    Neither ‘Bag’, ‘BasicTypes’, ‘BooleanFormula’, ‘Class’, ‘CoAxiom’,
            ‘FastString’, ‘ForeignCall’, ‘GHC’, ‘Name’,
            ‘Outputable’ nor ‘RdrName’ exports ‘XConDecl’.
     |
2717 |   markAST _ (GHC.XConDecl _)
     |              ^^^^^^^^^^^^
cabal: Failed to build ghc-exactprint-0.5.8.1 (which is required by
exe:hie-wrapper from haskell-ide-engine-0.2.2.0 and exe:hie from
haskell-ide-engine-0.2.2.0). See the build log above for details.
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -50,6 +50,10 @@ defaults: &defaults
         name: Generate Hoogle database
         command: stack --stack-yaml=${STACK_FILE} exec hoogle generate
 
+    - run:
+        name: Clear cabal-helper cache
+        command: rm -fr ~/.cache/cabal-helper
+
     - save_cache:
         key: stack-cache-10-{{ arch }}-{{ .Environment.CIRCLE_JOB }}-{{ checksum "resolver.txt" }}
         paths:

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -29,6 +29,8 @@ defaults: &defaults
     - run:
         name: Stack setup
         command: stack -j 2 --stack-yaml=${STACK_FILE} setup
+        name: Stack setup GHC 8.4.4, to make sure it can be used
+        command: stack -j 2 setup 8.4.4
 
     - run:
         name: Install happy

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -29,6 +29,7 @@ defaults: &defaults
     - run:
         name: Stack setup
         command: stack -j 2 --stack-yaml=${STACK_FILE} setup
+    - run:
         name: Stack setup GHC 8.4.4, to make sure it can be used
         command: stack -j 2 setup 8.4.4
 

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -124,6 +124,12 @@ jobs:
       - STACK_FILE: "stack-8.4.3.yaml"
     <<: *defaults
 
+  ghc-8.4.4:
+    environment:
+      - STACK_FILE: "stack-8.4.4.yaml"
+    <<: *defaults
+
+
   ghc-nightly:
     environment:
       - STACK_FILE: "stack.yaml"
@@ -170,5 +176,6 @@ workflows:
       - ghc-8.2.2
       - ghc-8.4.2
       - ghc-8.4.3
+      - ghc-8.4.4
       - ghc-nightly
       - cabal

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -22,9 +22,9 @@ defaults: &defaults
 
     - restore_cache:
         keys:
-          - stack-cache-09-{{ arch }}-{{ .Environment.CIRCLE_JOB }}-{{ checksum "stack-build.txt" }}-{{ checksum "all-cabal.txt" }}
-          - stack-cache-09-{{ arch }}-{{ .Environment.CIRCLE_JOB }}-{{ checksum "stack-build.txt" }}
-          - stack-cache-09-{{ arch }}-{{ .Environment.CIRCLE_JOB }}-{{ checksum "resolver.txt" }}
+          - stack-cache-10-{{ arch }}-{{ .Environment.CIRCLE_JOB }}-{{ checksum "stack-build.txt" }}-{{ checksum "all-cabal.txt" }}
+          - stack-cache-10-{{ arch }}-{{ .Environment.CIRCLE_JOB }}-{{ checksum "stack-build.txt" }}
+          - stack-cache-10-{{ arch }}-{{ .Environment.CIRCLE_JOB }}-{{ checksum "resolver.txt" }}
 
     - run:
         name: Stack setup
@@ -51,7 +51,7 @@ defaults: &defaults
         command: stack --stack-yaml=${STACK_FILE} exec hoogle generate
 
     - save_cache:
-        key: stack-cache-09-{{ arch }}-{{ .Environment.CIRCLE_JOB }}-{{ checksum "resolver.txt" }}
+        key: stack-cache-10-{{ arch }}-{{ .Environment.CIRCLE_JOB }}-{{ checksum "resolver.txt" }}
         paths:
           - ~/.stack
           - ~/.cache
@@ -72,7 +72,7 @@ defaults: &defaults
         path: test-logs
 
     - save_cache:
-        key: stack-cache-09-{{ arch }}-{{ .Environment.CIRCLE_JOB }}-{{ checksum "stack-build.txt" }}
+        key: stack-cache-10-{{ arch }}-{{ .Environment.CIRCLE_JOB }}-{{ checksum "stack-build.txt" }}
         paths:
           - ~/.stack
           - ~/.cache
@@ -85,7 +85,7 @@ defaults: &defaults
           - ~/build/submodules/cabal-helper/.stack-work
 
     - save_cache:
-        key: stack-cache-09-{{ arch }}-{{ .Environment.CIRCLE_JOB }}-{{ checksum "stack-build.txt" }}-{{ checksum "all-cabal.txt" }}
+        key: stack-cache-10-{{ arch }}-{{ .Environment.CIRCLE_JOB }}-{{ checksum "stack-build.txt" }}-{{ checksum "all-cabal.txt" }}
         paths:
           - ~/.stack
           - ~/.cache

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -29,9 +29,6 @@ defaults: &defaults
     - run:
         name: Stack setup
         command: stack -j 2 --stack-yaml=${STACK_FILE} setup
-    - run:
-        name: Stack setup GHC 8.4.4, to make sure it can be used
-        command: stack -j 2 setup 8.4.4
 
     - run:
         name: Install happy

--- a/Makefile
+++ b/Makefile
@@ -4,8 +4,8 @@ STACKLOCALBINDIR:=$(shell stack path --local-bin)
 all: help
 .PHONY: all
 
-## Builds hie for all supported GHC versions (8.2.1, 8.2.2, 8.4.2 and 8.4.3)
-build: hie-8.2.1 hie-8.2.2 hie-8.4.2 hie-8.4.3
+## Builds hie for all supported GHC versions (8.2.1, 8.2.2, 8.4.2 and 8.4.3, 8.4.4)
+build: hie-8.2.1 hie-8.2.2 hie-8.4.2 hie-8.4.3 hie-8.4.4
 .PHONY: build
 
 ## Builds hie and hoogle databases for all supported GHC versions
@@ -47,6 +47,14 @@ hie-8.4.3: submodules
 		&& cp '$(STACKLOCALBINDIR)/hie' '$(STACKLOCALBINDIR)/hie-8.4.3'    \
 		&& cp '$(STACKLOCALBINDIR)/hie-8.4.3' '$(STACKLOCALBINDIR)/hie-8.4'
 .PHONY: hie-8.4.3
+
+## Builds hie for GHC version 8.4.4 only
+hie-8.4.4: submodules
+	stack --stack-yaml=stack-8.4.4.yaml build
+	stack --stack-yaml=stack-8.4.4.yaml install                                      \
+		&& cp '$(STACKLOCALBINDIR)/hie' '$(STACKLOCALBINDIR)/hie-8.4.4'    \
+		&& cp '$(STACKLOCALBINDIR)/hie-8.4.4' '$(STACKLOCALBINDIR)/hie-8.4'
+.PHONY: hie-8.4.4
 
 # ------------------------------------------------------
 

--- a/stack-8.4.3.yaml
+++ b/stack-8.4.3.yaml
@@ -1,4 +1,4 @@
-resolver: lts-12.11
+resolver: lts-12.14 # Last for GHC 8.4.3
 packages:
 - .
 - hie-plugin-api

--- a/stack-8.4.4.yaml
+++ b/stack-8.4.4.yaml
@@ -1,5 +1,4 @@
-resolver: lts-12.14
-compiler-check: newer-minor
+resolver: lts-12.15 # LTS 12.15 is first to support GHC 8.4.4
 packages:
 - .
 - hie-plugin-api

--- a/stack-8.4.4.yaml
+++ b/stack-8.4.4.yaml
@@ -1,4 +1,5 @@
-resolver: lts-12.11
+resolver: lts-12.14
+compiler-check: newer-minor
 packages:
 - .
 - hie-plugin-api

--- a/test/utils/TestUtils.hs
+++ b/test/utils/TestUtils.hs
@@ -129,8 +129,10 @@ ghc84 = False
 
 stackYaml :: FilePath
 stackYaml =
-#if (defined(MIN_VERSION_GLASGOW_HASKELL) && (MIN_VERSION_GLASGOW_HASKELL(8,4,3,0)))
-  "stack.yaml"
+#if (defined(MIN_VERSION_GLASGOW_HASKELL) && (MIN_VERSION_GLASGOW_HASKELL(8,4,4,0)))
+  "stack-8.4.4.yaml"
+#elif (defined(MIN_VERSION_GLASGOW_HASKELL) && (MIN_VERSION_GLASGOW_HASKELL(8,4,3,0)))
+  "stack-8.4.3.yaml"
 #elif (defined(MIN_VERSION_GLASGOW_HASKELL) && (MIN_VERSION_GLASGOW_HASKELL(8,4,2,0)))
   "stack-8.4.2.yaml"
 #elif (defined(MIN_VERSION_GLASGOW_HASKELL) && (MIN_VERSION_GLASGOW_HASKELL(8,2,2,0)))

